symfony/plugins/orangehrmRESTPlugin/lib/Api/User/EmployeePunchInAPI.php

```diff
@@ -0,0 +1,142 @@
+            throw new InvalidParamException('Cannot Proceed Punch In Employee Already Punched In');
+        }
+        $attendanceRecord = new AttendanceRecord();
+        $attendanceRecord->setEmployeeId($empNumber);
+
+        $nextState = PluginAttendanceRecord::STATE_PUNCHED_IN;
+        if (empty($timeZone)) {
+            throw new InvalidParamException('TimeZone Cannot Be Empty');
+        }
+        if (!$this->getAttendanceService()->validateTimezone($timeZone)) {
+            throw new InvalidParamException('Invalid Time Zone');
+        }
+        $timeZoneDTZ = new DateTimeZone($timeZone);
+        $originDateTime = new DateTime($dateTime, $timeZoneDTZ);
+        $punchIndateTime = $originDateTime->format('Y-m-d H:i');
+        $timeZoneOffset = $this->getTimezoneOffset('UTC', $timeZone);
+
+        //check overlapping
+        $punchInUtcTime = $this->getAttendanceService()->getCalculatedPunchInUtcTime($punchIndateTime, $timeZoneOffset);
+        $isValid = $this->getAttendanceService()->checkForPunchInOverLappingRecords(
+            $punchInUtcTime,
+            $empNumber
+        );
+        if (!$isValid) {
+            throw new InvalidParamException('Overlapping Records Found');
+        }
+
+        try {
+            $attendanceRecord = $this->setPunchInRecord(
+                $attendanceRecord,
+                $nextState,
+                $punchInUtcTime,
+                $punchIndateTime,
+                $timeZoneOffset / 3600,
+                $punchInNote
+            );
+
+            $displayTimeZoneOffset = $this->getAttendanceService()->getOriginDisplayTimeZoneOffset(
+                $timeZoneOffset / 3600
+            );
+
+            return new Response(
+                array(
+                    'success' => 'Successfully Punched In',
+                    'id' => $attendanceRecord->getId(),
+                    'datetime' => $attendanceRecord->getPunchInUserTime(),
+                    'timezone' => $displayTimeZoneOffset,
+                    'note' => $attendanceRecord->getPunchInNote()
+                )
+            );
+        } catch (Exception $e) {
+            new BadRequestException($e->getMessage());
+        }
+    }
+
+    public function filterParameters()
+    {
+        $filters = array();
+        $filters['timezone'] = $this->getRequestParams()->getPostParam(parent::PARAMETER_TIME_ZONE);
+        $filters['punchInNote'] = $this->getRequestParams()->getPostParam(parent::PARAMETER_NOTE);
+        $filters['datetime'] = $this->getRequestParams()->getPostParam(parent::PARAMETER_DATE_TIME);
+        return $filters;
+    }
+
+    /**
+     * @return array
+     */
+    public function getValidationRules(): array
+    {
+        return [
+            parent::PARAMETER_NOTE => ['StringType' => true, 'Length' => [1, 250]],
+            parent::PARAMETER_DATE_TIME => ['Date' => ['Y-m-d H:i']],
+            parent::PARAMETER_TIME_ZONE => ['NotEmpty' => true,'StringType' => true]
```

Comment:
parent::PARAMETER_TIME_ZONE => ['NotEmpty' => true,'StringType' => true]

Validation should be improved. The provided Timezone value should be a value from a specific set of float values.
i.e :  array ( 0, 1.0, 2.0, 3.0, 4.0, 5.0, 5.5, 6.0, 7.0, 8.0, 9.0, 9.5, 10.0, 11.0, 12.0, -11.0, -10.0, -9.0, -8.0, -7.0, -6.0, -5.0, -4.0, -3.5, -3.0, -2.0, -1.0).
As per this validation rule, it accepts any string value.